### DISCUSSION TITLE
Should the origin (x=0, y=0) remain in lower-left hand corner?

### DISCUSSION THREAD

**Discussion by dusk125** [gopxl/pixel#8] [Polls]
PIxel is unique in its placement of its origin: most graphics libraries/engines have the origin point in the upper-left hand corner of the screen and down is positive movement in the y axis, Pixel has its origin at the lower-left hand corner and up is the positive y-axis movement.

This creates annoying issues when integrating with other libraries (like imgui) or saving screenshot or doing other image manipulation.

**Comment by bhperry** [comment]
Voting both contingent on it not being a complete mess to implement and support that.

**Reply by MarkKremer** [reply]
Having the origin in the bottom left corner is how OpenGL works. I would leave the flipping to the very end or you'll end up writing `1-y` everywhere.

Edit: or change the camera matrix or something. I don't know how pixel works.

**Reply by dusk125** [reply]
I could've sworn that it was Pixel being wonky...but I'm wrong when I'm wrong.

Maybe the flipping code I'm remembering is for drawing images where they would be loaded too-left down.

**Reply by bhperry** [reply]
Funny enough, the exact oposite conversation has been going on a long time over on Godot engine https://github.com/godotengine/godot-proposals/issues/2048

**Reply by dusk125** [reply]
Huh, then maybe this is a nonissue and we should leave this be.
